src/service/url-replacements-impl.js

```diff
@@ -757,6 +777,23 @@ export class UrlReplacements {
     // FOO_BAR(arg1,arg2)
     return new RegExp('\\$?(' + all + ')(?:\\(([0-9a-zA-Z-_.,]+)\\))?', 'g');
   }
+
+  /**
+   * Ensures that the protocol of the original url matches the protocol of the
+   * replacement url. Returns the replacement if they do, the original if they
+   * do not.
+   * @param {string} url
+   * @param {string} replacement
+   * @return {string}
+   */
+  ensureProtocolMatches_(url, replacement) {
+    if (protocolOfUrl(replacement) !== protocolOfUrl(url)) {
```

Comment:
I'd also like to do an extra assert that the protocol is not `javascript:`. Not the only bad one, but a good defense in depth.